ocaml/xapi/xapi_xenops.ml

```diff
@@ -1687,7 +1687,10 @@ module Xenopsd_metadata = struct
 
   (* Unregisters a VM with xenopsd, and cleans up metadata and caches *)
   let pull ~__context id =
-    with_lock metadata_m (fun () ->
+    (* Continue even if mutex is already locked one level above *)
```

Comment:
Is it better to use a reentrant lock here?
I think Colin has done something before to make the db lock reentrant (https://github.com/xapi-project/xen-api/commit/fae9b559cf40306a2a0a6ac2a11e10a4c7f94572), could we borrow that or use a library?